interfaces/tests/file_int/file_int.c

```diff
@@ -0,0 +1,427 @@
+/*Tests_SRS_FILE_43_001 : [file_create shall open the file named full_file_name for asynchronous operationsand return its handle.]*/
+/*Tests_SRS_FILE_43_006 : [file_destroy shall wait for all pending I / O operations to complete.]*/
+/*Tests_SRS_FILE_43_007 : [file_destroy shall close the file handle handle.]*/
+/*Tests_SRS_FILE_43_014 : [file_write_async shall enqueue a write request to write source's content to the position offset in the file. ]*/
+/*Tests_SRS_FILE_43_041:[If position + size is greater than the size of the file and the call to write is succesfull, file_write_async shall grow the file to accomodate the write.]*/
+/*Tests_SRS_FILE_43_008 : [file_write_async shall call user_call_back passing user_context and success depending on the success of the asynchronous write operation.]*/
+/*Tests_SRS_FILE_43_030 : [file_write_async shall succeed and return FILE_WRITE_ASYNC_OK.]*/
+/*Tests_SRS_FILE_43_021 : [file_read_async shall enqueue a read request to read handle's content at position offset and write it to destination. ]*/
+/*Tests_SRS_FILE_43_016:[file_read_async shall call user_callback passing user_context and success depending on the success of the asynchronous read operation.]*/
+/*Tests_SRS_FILE_43_031 : [file_read_async shall succeed and return FILE_READ_ASYNC_OK.]*/
+TEST_FUNCTION(write_twice_to_a_file_contiguously_and_read_from_it)
+{
+    ///arrange
+    const int size = 5;
+
+    unsigned char source1[] = "abcd";
+    WRITE_COMPLETE_CONTEXT write_context1;
+    write_context1.pre_callback_value = 41;
+    interlocked_exchange(&write_context1.value, write_context1.pre_callback_value);
+    write_context1.post_callback_value = 42;
+
+    unsigned char source2[] = "efgh";
+    WRITE_COMPLETE_CONTEXT write_context2;
+    write_context2.pre_callback_value = 45;
+    interlocked_exchange(&write_context2.value, write_context2.pre_callback_value);
+    write_context2.post_callback_value = 46;
+
+    unsigned char destination[9];
+    READ_COMPLETE_CONTEXT read_context;
+    read_context.pre_callback_value = 43;
+    interlocked_exchange(&read_context.value, read_context.pre_callback_value);
+    read_context.post_callback_value = 44;
+
+
+    EXECUTION_ENGINE_HANDLE execution_engine = execution_engine_create(NULL);
+    FILE_HANDLE file_handle = file_create(execution_engine, "write_twice_to_a_file_contiguously_and_read_from_it.txt", NULL, NULL);
+
+    ///act
+    FILE_WRITE_ASYNC_RESULT write_result1 = file_write_async(file_handle, source1, size, 0, write_callback, &write_context1);
+    wait_on_address(&write_context1.value, &write_context1.pre_callback_value, UINT32_MAX);
+    FILE_WRITE_ASYNC_RESULT write_result2 = file_write_async(file_handle, source2, size, 4, write_callback, &write_context2);
+    wait_on_address(&write_context2.value, &write_context2.pre_callback_value, UINT32_MAX);
+    FILE_READ_ASYNC_RESULT read_result = file_read_async(file_handle, destination, 9, 0, read_callback, &read_context);

```

Comment:
>9 [](start = 83, length = 1)

I'd always use `sizeof(destination)` when you have a stack allocated array, because then you don't forget to update this if the test is changed. Here and everywhere.

Reply:
done

---
In reply to: [456715370](https://github.com/Azure/azure-c-pal/pull/10#discussion_r456715370) [](ancestors = 456715370)